pandas/core/groupby/generic.py

```diff
@@ -401,7 +401,8 @@ def _wrap_applied_output(
                 not_indexed_same=not_indexed_same,
                 is_transform=is_transform,
             )
-            result.name = self.obj.name
+            if isinstance(result, Series):
```

Comment:
Can you use `result.ndim == 1` here instead.

Reply:
sure, will do - though just for my understanding, why?

Reply:
Interesting - it used to be that this instance checks were quite a bit more expensive than checking the `ndim` attribute; I'm now seeing isinstance is slightly cheaper. With type checkers able to naturally derive type information from isinstance checks in code paths, we should certainly prefer isinstance if they are just as performant.

Reply:
@MarcoGorelli - I'm good with either using ndim or isinstance here; I plan on taking a more in depth look of our usage. Let me know what you think.

Reply:
no preference from my end